Source/WebCore/display/css/DisplayBoxDecorationPainter.cpp

```diff
@@ -1260,7 +1260,19 @@ void BoxDecorationPainter::paintBoxShadow(PaintingContext& paintingContext, Shad
 
     bool hasOpaqueBackground = m_box.style().backgroundColor().isOpaque();
 
-    auto paintNormalShadow = [&](const ShadowData& shadow) {
+    auto resolveColor = [] (const StyleColor& styleColor) {
+        if (styleColor.isAbsoluteColor())
+            return styleColor.absoluteColor();
+
+        // FIXME: StyleColor should have been properly resolved already,
+        // but unfortunately it's not enforced in the type system.
+        // We should create a new class such as ResolvedShadowData to make it explicit.
+        // https://bugs.webkit.org/show_bug.cgi?id=248467
+        ASSERT_NOT_REACHED();
+        return Color { };
+    };
```

Comment:
Can we confidently do this instead?

```suggestion
    auto resolveColor = [] (const StyleColor& styleColor) {
        // FIXME: StyleColor should have been properly resolved already,
        // but unfortunately it's not enforced in the type system.
        // We should create a new class such as ResolvedShadowData to make it explicit.
        // https://bugs.webkit.org/show_bug.cgi?id=248467
        ASSERT(styleColor.isAbsoluteColor());
        return styleColor.absoluteColor();
    };
```

Reply:
In release build, this would be UB whereas the `Color { }` is not ; so it's kind of a safety net.

Reply:
This isn't shipping code, but an experiment, so we shouldn't care about release builds too much. I'd rather have it crash so we catch issues with the design.

Reply:
Changed